calico/felix/test/test_config.py

```diff
@@ -273,4 +285,117 @@ def test_ip_in_ip_enabled_bad(self):
+        with mock.patch('calico.common.complete_logging'):
+            config = Config("calico/felix/test/data/felix_missing.cfg")
+        cfg_dict = { "InterfacePrefix": "blah",
+                     "ReportingTTLSecs": "NaN"}
+        with self.assertRaisesRegexp(ConfigException,
+                                     "Field was not integer.*"):
+            config.report_etcd_config({}, cfg_dict)
+
+    def test_reporting_interval_not_int(self):
+        """
+        Test exception is rased ifstatus reporting interval has invalid (non-integer) value.
+        """
+        with mock.patch('calico.common.complete_logging'):
+            config = Config("calico/felix/test/data/felix_missing.cfg")
+        cfg_dict = { "InterfacePrefix": "blah",
+                     "ReportingIntervalSecs": "NaN"}
+        with self.assertRaisesRegexp(ConfigException,
+                                     "Field was not integer.*"):
+            config.report_etcd_config({}, cfg_dict)
+
+    def test_negative_reporting_interval(self):
+        """
+        Test that status reporting is disabled if interval time is negative.
+        """
+        with mock.patch('calico.common.complete_logging'):
+            config = Config("calico/felix/test/data/felix_missing.cfg")
+        cfg_dict = { "InterfacePrefix": "blah",
+                     "ReportingIntervalSecs": -42,
+                     "ReportingTTLSecs": 7 }
+        with mock.patch('calico.common.complete_logging'):
+            config.report_etcd_config({}, cfg_dict)
+
+        self.assertEqual(config.REPORTING_INTERVAL_SECS, 0)
+        self.assertEqual(config.REPORTING_TTL_SECS, 0)
+
+    def test_default_ttl(self):
+        """
+        Test that ttl is set to 2.5 times interval for default ttl.
+        """
+        with mock.patch('calico.common.complete_logging'):
+            config = Config("calico/felix/test/data/felix_missing.cfg")
+        cfg_dict = { "InterfacePrefix": "blah",
+                     "ReportingIntervalSecs": "21" }
+        with mock.patch('calico.common.complete_logging'):
+            config.report_etcd_config({}, cfg_dict)
+
+        self.assertEqual(config.REPORTING_TTL_SECS, 52)
+
+    def test_valid_interval_and_ttl(self):
+        """
+        Test valid reporting parameters are not chaned.
+        """
+        with mock.patch('calico.common.complete_logging'):
+            config = Config("calico/felix/test/data/felix_missing.cfg")
+        cfg_dict = { "InterfacePrefix": "blah",
+                     "ReportingIntervalSecs": 42,
+                     "ReportingTTLSecs": 47}
+        with mock.patch('calico.common.complete_logging'):
+            config.report_etcd_config({}, cfg_dict)
+
+        self.assertEqual(config.REPORTING_INTERVAL_SECS, 42)
+        self.assertEqual(config.REPORTING_TTL_SECS, 47)
+
+    def test_reporting_ttl_less_than_interval(self):
+        """
+        Test that exception is raised when status reporting ttl is less than interval.
```

Comment:
I think this docstring is a bit misleading &ndash; this test doesn't check if an exception is raised, and the relevant code doesn't raise any exceptions. We're testing that the ttl is defaulted correctly if it's less than the interval.